### ISSUE TITLE
Custom Modifiers Broken ?

### ISSUE THREAD

**Issue by BerndWessels** [popperjs/react-popper#362]
Hello

Adding any `custom modifier` with a `fn` to the `modifiers` completely breaks the browser - like something seems to go into an infinite loop, even preventing the page from rendering and no errors in the console.

Here is an example, but even much simpler custom modifiers will cause the problem as soon as you add a `fn` to it.

```js
import React, { forwardRef } from "react";
import PropTypes from "prop-types";
import styled from "styled-components/macro";
import { usePopper } from "react-popper";
import useForwardRef from "../../../utils/use-forward-ref";

const Container = styled.div`
  z-index: 1;
  background-color: ${({ theme }) => theme.palette.regular.main};
  border-color: ${({ theme }) => theme.palette.regular.border};
  border-radius: ${({ theme }) => theme.form.components.borderRadius};
  border-style: ${({ theme }) => theme.borders.regular.style};
  border-width: ${({ theme }) => theme.borders.regular.width}px;
  box-shadow: ${({ theme }) => theme.shadows[5]};
  box-sizing: border-box;
  display: ${({ show }) => (show ? "block" : "none")};
`;

const Component = forwardRef(({ children, targetRef, ...props }, ref) => {
  const popperRef = useForwardRef(ref);
  const { styles, attributes } = usePopper(
    targetRef ? targetRef.current : null,
    popperRef.current,
    {
      modifiers: [
        {
          name: "offset",
          options: {
            offset: [0, 4],
          },
        },
        {
          name: "sameWidth",
          enabled: true,
          phase: "beforeWrite",
          requires: ["computeStyles"],
          fn: ({ state }) => {
            state.styles.popper.width = `${state.rects.reference.width}px`;
          },
          effect: ({ state }) => {
            state.elements.popper.style.width = `${
                state.elements.reference.offsetWidth
            }px`;
          }
        }
      ],
      placement: "bottom-start",
    }
  );
  return (
    <Container ref={popperRef} style={styles.popper} {...attributes} {...props}>
      {children}
    </Container>
  );
});

Component.defaultProps = {
  children: undefined,
  targetRef: null,
  show: true,
};

Component.propTypes = {
  children: PropTypes.node,
  targetRef: PropTypes.oneOfType([
    // Either a function
    PropTypes.func,
    // Or the instance of a DOM native element (see the note about SSR)
    // eslint-disable-next-line no-undef
    PropTypes.shape({ current: PropTypes.instanceOf(Element) }),
  ]),
  show: PropTypes.bool,
};

Component.displayName = "Popover";

export default Component;
```

I'm using the latest versions:

```json
    "@popperjs/core": "^2.4.0",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-popper": "^2.2.3",
    "styled-components": "^5.1.0"
```

**Comment by FezVrasta** [comment]
Thank you for your issue.

Unfortunately I can't answer to all the reported issues without a little help from you.
Please, make sure to follow the **issues template** when you submit a new issue, in this way I will already have all the informations needed to understand the problem and help you with it.

I'm closing this issue for now, but feel free to open a new one making sure to follow all the instructions!

Thank you for your understanding.

**Comment by sajcics** [comment]
The same situation happens for me too. The same custom modifier had worked in popperjs/core implementation, and goes to infinite loop

**Comment by FezVrasta** [comment]
The same requirement applies, without a proper repro and the issue template followed I can't help anybody.
